web/search.dart

```diff
@@ -251,21 +292,71 @@ void initializeSearch(
   }
 
   void showSuggestions() {
-    if (presentation.hasChildNodes()) {
+    if (searchResults.hasChildNodes()) {
       listBox.style.display = 'block';
       listBox.setAttribute('aria-expanded', 'true');
     }
   }
 
+  // Function that iterates through the categoriesMap and appends it to the given HTML element
+  // The received Element is the one used for search results page or dropdown list in the search bar
+  void iterateCategoriesMap(Element e) {
+    categoriesMap.forEach((keys, values) {
+      var value = categoriesMap[keys];
+      e.append(value!);
+    });
+  }
+
+  // Function that creates the content displayed in the main-content element.
+  void searchResultPage(String input) {
+    var mainContent = document.getElementById('dartdoc-main-content');
+
+    if (mainContent == null) {
+      return;
+    }
+
+    mainContent.text = '';
+
+    var section = document.createElement('section');
+    section.classes.add('search-summary');
+    mainContent.append(section);
+
+    var title = document.createElement('h2');
+    title.innerHtml = 'Search Results';
+    mainContent.append(title);
+
+    var summary = document.createElement('div')
+      ..classes.add('search-summary')
+      ..innerHtml = '$suggestionLength results for "$input"';
+    mainContent.append(summary);
+
+    if (categoriesMap.isNotEmpty) {
+      iterateCategoriesMap(mainContent);
+    } else {
+      var noResults = document.createElement('div');
+      noResults.classes.add('search-summary');
+      noResults.innerHtml =
+          'There was not a match for "$input". Please try another search.';
+      mainContent.append(noResults);
+    }
+  }
+
   void hideSuggestions() {
     listBox.style.display = 'none';
     listBox.setAttribute('aria-expanded', 'false');
   }
 
+  void enterMessage() {
```

Comment:
Since this changes the page, I think an action-y verb in the name would be good, like "showEnterMessage" or something similar.